fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/CreateRDFSourcePersister.java

```diff
@@ -56,8 +57,19 @@ public void persist(final OCFLPersistentStorageSession session, final ResourceOp
 
         final CreateResourceOperation createResourceOp = ((CreateResourceOperation)operation);
         final boolean archivalGroup = createResourceOp.isArchivalGroup();
-        final String rootObjectId = archivalGroup ? createResourceOp.getResourceId() :
-                                                    resolveRootObjectId(createResourceOp, session);
+
+        final String rootObjectId;
+        if (archivalGroup) {
+            //if archival group, ensure that there are no archival group ancestors
+            if (findArchivalGroupInAncestry(resourceId, session) != null) {
```

Comment:
Data model validation feels like it might make sense to take place at the kernel level rather than persistence impl. It would span multiple persistence impls and would potentially be much cheaper to validate since we could check against indexes more easily, and likely already need other "is this parent okay?" checks, like preventing using a binary as a parent. In that case we would presumably also not have to catch persistence errors in fcrepo-http-commons.

Not sure if it makes sense to move it at this point or not, wanted to bring up the point though to see what you think

Reply:
I think it that makes sense:  ie that the check  take place in the kernel layer.  I would prefer that we address this issue in a separate JIRA as I think this will require some more thinking.   For the time being this works.